### ISSUE TITLE
Implement Tags API

### ISSUE THREAD

**Issue by tdstein** [posit-dev/posit-sdk-py#162]
# Tags API

## v0.6.0 Proposal

```python
from posit import connect

client = connect.Client()
content = client.content.get(...)
content.tags.find(...)
content.tags.find_one(...)
content.tags.get(...)
content.tags.count()

tag = content.tags.create(...)
tag.update(...)
tag.delete()
```

## v1.0.0 Proposal

```python
content.tags.all()
content.tags.find(...)
content.tags.find_by(...)
content.tags.where(...)

tag = content.tags.create(...)
tag.update(...)
tag.save()
tag.destroy()
```

**Comment by tdstein** [comment]
Update https://github.com/posit-dev/connect-cookbook/issues/47 with the implementation once complete.

**Comment by schloerke** [comment]
Possibly addressed by https://github.com/posit-dev/posit-sdk-py/pull/346

```python
content.tags.find()
content.tags.add(...)
content.tags.delete(...)

child_tags = tag.child_tags.find()
desc_tags = tag.descendant_tags.find()
parent_tag = tag.parent_tag
content_items = tag.content_items.find()

tag.destroy()

# Missing in PR
updated_tag = tag.update(...)
```

**Comment by tdstein** [comment]
Resolved by #350 and #351
